test/support/utils.js

```diff
@@ -6,20 +6,29 @@ var request = require('supertest')
 
 exports.assert = assert
 exports.createHitHandle = createHitHandle
+exports.createErrorHitHandle = createErrorHitHandle
 exports.createServer = createServer
 exports.rawrequest = rawrequest
 exports.request = request
 exports.shouldHitHandle = shouldHitHandle
 exports.shouldNotHitHandle = shouldNotHitHandle
+exports.manyAsyncCalls = manyAsyncCalls
 
 function createHitHandle(num) {
   var name = 'x-fn-' + String(num)
   return function hit(req, res, next) {
-    res.setHeader(name, 'hit')
+    if (!(res.headersSent || res._headerSent)) { res.setHeader(name, 'hit') }
```

Comment:
Just use `if (res._header)` for the condition for full Node.js compatibility.

Even then, this kind of a change in a fundamental utility is not good and we don't want; it would hide errors silently.